pkg/helm/watcher/cache/cache.go

```diff
@@ -139,15 +141,10 @@ func (c *ProfileCache) ListProfiles(ctx context.Context, helmRepoNamespace, helm
 	var result []*pb.Profile
 
 	listOperation := func() error {
-		content, err := os.ReadFile(filepath.Join(c.cacheLocation, helmRepoNamespace, helmRepoName, profileFilename))
-		if err != nil {
+		if err := c.getProfilesFromFile(helmRepoNamespace, helmRepoName, &result); err != nil {
 			return fmt.Errorf("failed to read profiles data for helm repo: %w", err)
```

Comment:
Could this include the details of the Helm repo it couldn't read profiles data from?